### PR TITLE
bugfix/start-of-week patch

### PR DESCRIPTION
**Pull request by fivetran-joemarkiewicz**
**Are you a current Fivetran customer?** 

Fivetran created PR

**What change(s) does this PR introduce?** 

It was brought to our attention that the `dbt_utils.date_trunc` macro only leverages the default arguments of the date_trunc function in the various warehouses. For example, `date_trunc` in Snowflake for the `week` argument produces the starting Monday, while BigQuery produces the starting Sunday. For this package, we want to leverage the start of the week as Sunday. Therefore, logic within the business metric intermediate models has been adjusted to capture the start of the week as Sunday. This was done by leveraging the `week_start` macro within the `dbt-date` package.

**Did you update the CHANGELOG?** 

- [X] Yes

**Does this PR introduce a breaking change?**

- [ ] Yes (please provide breaking change details below.)
- [X] No  (please provide an explanation as to how the change is non-breaking below.)

This will simply fix underlying date logic for business metrics and will not result in a breaking change.

**Did you update the dbt_project.yml files with the version upgrade (please leverage standard semantic versioning)? (In both your main project and integration_tests)** 

- [X] Yes

**Is this PR in response to a previously created Bug or Feature Request**

- [X] Yes, this was an internal issue that was raised
- [ ] No 

**How did you test the PR changes?** 

 

- [X] CircleCi  
- [X] Local (please provide additional testing details below)

These changes were tested with the user who raised the issue and we were able to see the business metrics tie out accurately with the new changes.

**Select which warehouse(s) were used to test the PR**
 

- [X] BigQuery
- [X] Redshift
- [X] Snowflake
- [X] Postgres
- [ ] Databricks
- [ ] Other (provide details below)

**Provide an emoji that best describes your current mood**
 
⌚ 

**Feedback**

We are so excited you decided to contribute to the Fivetran community dbt package! We continue to work to improve the packages and would greatly appreciate your [feedback](https://www.surveymonkey.com/r/DQ7K7WW) on our existing dbt packages or what you'd like to see next.

### CODE REVIEW

File: models/agent_work_time/int_zendesk__ticket_work_time_business.sql [hunk 1]
```diff
@@ -39,11 +39,11 @@ with ticket_historical_status as (
 
     select 
       ticket_status_crossed_with_schedule.*,
-      ({{ fivetran_utils.timestamp_diff(
-              "" ~ dbt_utils.date_trunc('week', 'ticket_status_crossed_with_schedule.status_schedule_start') ~ "", 
-              'ticket_status_crossed_with_schedule.status_schedule_start',
-              'second') }} /60
-            ) as start_time_in_minutes_from_week,
+    ({{ fivetran_utils.timestamp_diff(
+            "cast(" ~ dbt_date.week_start('ticket_status_crossed_with_schedule.status_schedule_start','UTC') ~ "as " ~ dbt_utils.type_timestamp() ~ ")",
```
Comment (fivetran-joemarkiewicz) [inline models/agent_work_time/int_zendesk__ticket_work_time_business.sql:43]: Both fields needed to be cast as timestamp or else the calculations would be off.

**Review by fivetran-jamie — APPROVED** [review]
looks good!!

File: packages.yml [hunk 2]
```diff
@@ -1,3 +1,5 @@
 packages:
   - package: fivetran/zendesk_source
-    version: [">=0.6.0","<0.7.0"] 
\ No newline at end of file
+    version: [">=0.6.0","<0.7.0"] 
+  - package: calogica/dbt_date
```
[the remark this answers is not in the document]
Comment (fivetran-joemarkiewicz) [reply packages.yml:4]: Good call!

File: models/agent_work_time/int_zendesk__ticket_work_time_business.sql (hunk 1 above)
Comment (fivetran-joemarkiewicz) [reply models/agent_work_time/int_zendesk__ticket_work_time_business.sql:43]: @fivetran-jamie exactly!

File: models/reply_times/int_zendesk__ticket_first_reply_time_business.sql [hunk 3]
```diff
@@ -31,14 +32,15 @@ with ticket_reply_times as (
     first_reply_time.ticket_id,
     ticket_schedules.schedule_created_at,
     ticket_schedules.schedule_invalidated_at,
+    cast({{ dbt_date.week_start("ticket_schedules.schedule_created_at","UTC") }} as {{ dbt_utils.type_timestamp() }}) as week_start,
```
[the remark this answers is not in the document]
Comment (fivetran-joemarkiewicz) [reply models/reply_times/int_zendesk__ticket_first_reply_time_business.sql]: Removed

**Pull request merged**